static/js/admin2.js

```diff
@@ -195,66 +195,96 @@ const addTableRows = (rows) => {
-    const search = document.getElementById("search").value;
+const applyFilters = async () => {
+    console.log("applyFilters called"); // Debugging
+    const searchQuery = document.getElementById("search").value;
+    console.log("Search query:", searchQuery); // Debugging
     
-    console.log("Applying filters:", { sortBy, sortOrder, membership, semesters, search });
-    
-    const queryParams = new URLSearchParams();
-    queryParams.append("filter-sort-by", sortBy);
-    queryParams.append("filter-sort-order", sortOrder);
-    queryParams.append("filter-membership", membership);
-    queryParams.append("filter-semesters", semesters);
-    if (search) {
-        queryParams.append("search", search);
+    // Build filter parameters
+    const filterParams = new URLSearchParams();
+    if (searchQuery) {
+        filterParams.append("search", searchQuery);
     }
     
-    const endpoint = `/admin/profiles/1?${queryParams.toString()}`;
-    
-    fetch(endpoint)
-        .then(response => {
-            if (!response.ok) {
-                throw new Error(`HTTP error! Status: ${response.status}`);
+    // Add other filter parameters from the form
+    const filterForm = document.getElementById("filter-settings");
+    if (filterForm) {
+        const formData = new FormData(filterForm);
+        for (let [key, value] of formData.entries()) {
+            if (value !== "All") {
+                filterParams.append(key, value);
             }
-            return response.json();
-        })
-        .then(data => {
-            const table = document.getElementById("dbTable");
-            
-            // Remove existing rows
-            table.querySelectorAll(".dbTableRow").forEach(row => row.remove());
-            addTableRows(data);
-        })
-        .catch(err => {
-            console.error("Filter error:", err);
-            alert("There was an error applying the filters. Please try again.");
-        });
+        }
+    }
+    
+    // Make the API request with filters
+    try {
+        const endpoint = `/admin/profiles/${getCurrentOrgId()}?${filterParams.toString()}`;
+        console.log("Filter endpoint:", endpoint); // Debugging
+        
+        const response = await fetch(endpoint);
+        if (!response.ok) {
+            throw new Error(`Response status: ${response.status}`);
+        }
+        
+        const data = await response.json();
+        console.log("Filter results:", data.length); // Debugging
+        
+        const table = document.getElementById("dbTable");
+        
+        // Remove existing rows
+        table.querySelectorAll(".dbTableRow").forEach(row => row.remove());
+        addTableRows(data);
+    }
+    catch (err) {
+        console.error("Filter error:", err);
+        alert("There was an error applying the filters. Please try again.");
+    }
 };
 
+// Helper function to get organization ID
+function getCurrentOrgId() {
```

Comment:
I think I set it up to include the org ID in the body tag as a data attribute. I think that's the cleanest way to go about this.